docs/tutorials/mongoid-documents.txt

```diff
@@ -176,6 +177,31 @@ Types that are not supported as dynamic attributes since they cannot be cast are
 - ``DateTime``
 - ``Range``
 
+The StringifiedSymbol Type
+--------------------------
+
+This type allows users to write Ruby code that makes use of the built-in Ruby Symbol type,
+while ensuring interoperability with other drivers. The StringifiedSymbol type stores all data
+on the MongoDB backend as a String, while returning values to the user as a Symbol.
+
+For example, when setting a field to a string value:
+
+.. code-block:: ruby
+
+  class Post
+    include Mongoid::Document
+
+    field :status, type: StringifiedSymbol
+  end
+
+  post = Post.new(status: "hello")
+  post.status # => :hello
+
+All values will be stringified upon being sent to the database, and all values will be converted
+to symbols when returned to the user. So, in the above example, setting ``:status`` to ``42``
+would result in ``"42"`` being stored in the database, and ``:"42"``
+being returned to the user.
+
```

Comment:
Two more things that I'd love to see in this documentation:
1. Could you add a sentence or two about the fact that this type will lazily migrate any BSON::Symbols already stored in the database to BSON::Strings when the user updates the document being stored? (Let me know if you have questions about how this works.)
2. Could you add a link to the [BSON Symbol Type](https://docs.mongodb.com/mongoid/current/tutorials/mongoid-documents/#bson-symbol-type) section of the docs for anybody who wants to learn more about using BSON::Symbols with Mongoid?

Reply:
There are examples of how to link things throughout the docs but it's a bit finicky so let me know if you need any help with that.